packages/firestore/src/platform/platform.ts

```diff
@@ -43,6 +43,12 @@ export interface Platform {
   /** Converts a binary string to a Base64 encoded string. */
   btoa(raw: string): string;
 
+  /**
+   * Generates `nBytes` of random bytes. If `nBytes` is negative, an empty array
+   * will be returned.
```

Comment:
What is the rationale for accepting negative numbers? This seems like it may be hiding bugs in the calling code.

Reply:
Changed it to validate input, and reject anything <=0.